renderdoc/driver/vulkan/wrappers/vk_shader_funcs.cpp

```diff
@@ -1464,14 +1481,37 @@ VkResult WrappedVulkan::vkCreateRayTracingPipelinesKHR(
 {
   VkResult ret;
 
-  // to be extra sure just in case the driver doesn't, set pipelines to VK_NULL_HANDLE first.
+  VkRayTracingPipelineCreateInfoKHR *unwrappedCreateInfos =
+      UnwrapInfos(m_State, pCreateInfos, createInfoCount);
+
   for(uint32_t i = 0; i < createInfoCount; i++)
+  {
+    // to be extra sure just in case the driver doesn't, set pipelines to VK_NULL_HANDLE first.
     pPipelines[i] = VK_NULL_HANDLE;
 
+    // Patch in capture/replay creation flags
+    VkBaseOutStructure *nextStruct = (VkBaseOutStructure *)unwrappedCreateInfos[i].pNext;
+    while(nextStruct)
+    {
+      if(nextStruct->sType != VK_STRUCTURE_TYPE_PIPELINE_CREATE_FLAGS_2_CREATE_INFO_KHR)
+      {
+        nextStruct = nextStruct->pNext;
+        continue;
+      }
+      VkPipelineCreateFlags2CreateInfoKHR *flagsInfo =
+          (VkPipelineCreateFlags2CreateInfoKHR *)nextStruct;
+      flagsInfo->flags |= VK_PIPELINE_CREATE_2_RAY_TRACING_SHADER_GROUP_HANDLE_CAPTURE_REPLAY_BIT_KHR;
+      break;
+    }
+    if(!nextStruct)
+      unwrappedCreateInfos[i].flags |=
+          VK_PIPELINE_CREATE_RAY_TRACING_SHADER_GROUP_HANDLE_CAPTURE_REPLAY_BIT_KHR;
+  }
+
   // deferred operations are currently not wrapped
   SERIALISE_TIME_CALL(ret = ObjDisp(device)->CreateRayTracingPipelinesKHR(
                           Unwrap(device), VK_NULL_HANDLE, Unwrap(pipelineCache), createInfoCount,
-                          UnwrapInfos(m_State, pCreateInfos, createInfoCount), NULL, pPipelines));
```

Comment:
Specifically below here where we already have `modifiedCreateInfo` you'd want to apply a similar kind of patching to add the necessary flag.

Reply:
I started on this, but it's actually a bit more complicated than the existing `modifiedCreateInfo` patching. We need to patch any `VkPipelineCreateFlags2CreateInfoKHR` if it exists, which by extension means we need to patch the entire `pNext` chain (or at least, all elements of it up until `VkPipelineCreateFlags2CreateInfoKHR`). I'm not sure how to go about that properly?

For what it's worth, it seems like the current code is buggy in that regard since the derivative pipeline check also only considers the flags from the pipeline create info itself, not of any potential `VkPipelineCreateFlags2CreateInfoKHR`.

Reply:
You actually make a very good point about the derivative check. It made me go to check - that flags2 struct is added in KHR_maintenance5 which RenderDoc does not currently support, the existing code was copied from pre-existing handling on normal pipelines.

So none of this handling is needed yet, and the code is not currently buggy (just lacking support). The new code you've added is fine to sttay as it's illegal for any application to do pass the struct in anyway, and it will be correct for the future.

Unfortunately the next-chain pattern in general is extremely annoying when it comes to patching... I have helpers for copying a whole chain to patch it but it gets messy quickly.